equinix/resource_fabric_network.go

```diff
@@ -0,0 +1,431 @@
+		"connections_count": nt.ConnectionsCount,
+	})
+	if err != nil {
+		return diag.FromErr(equinix_errors.FormatFabricError(err))
+	}
+	return diags
+}
+func getFabricNetworkUpdateRequest(network v4.Network, d *schema.ResourceData) (v4.NetworkChangeOperation, error) {
+	changeOps := v4.NetworkChangeOperation{}
+	existingName := network.Name
+	updateNameVal := d.Get("name")
+
+	log.Printf("existing name %s, Update Name Request %s ", existingName, updateNameVal)
+
+	if existingName != updateNameVal {
+		changeOps = v4.NetworkChangeOperation{Op: "replace", Path: "/name", Value: &updateNameVal}
+	} else {
+		return changeOps, fmt.Errorf("nothing to update for the Fabric Network: %s", existingName)
+	}
+	return changeOps, nil
+}
+func resourceFabricNetworkUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.Config).FabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	dbConn, err := waitUntilFabricNetworkIsProvisioned(d.Id(), meta, ctx)
+	if err != nil {
+		if !strings.Contains(err.Error(), "500") {
+			d.SetId("")
+		}
+		return diag.Errorf("either timed out or errored out while fetching Fabric Network for uuid %s and error %v", d.Id(), err)
+	}
+	// TO-DO
+	update, err := getFabricNetworkUpdateRequest(dbConn, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updates := []v4.NetworkChangeOperation{update}
+	_, res, err := client.NetworksApi.UpdateNetworkByUuid(ctx, updates, d.Id())
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error response for the Fabric Network update, response %v, error %v", res, equinix_errors.FormatFabricError(err)))
+	}
+	updateFg := v4.Network{}
+	updateFg, err = waitForFabricNetworkUpdateCompletion(d.Id(), meta, ctx)
+
+	if err != nil {
+		if !strings.Contains(err.Error(), "500") {
+			d.SetId("")
+		}
+		return diag.FromErr(fmt.Errorf("errored while waiting for successful Fabric Network update, response %v, error %v", res, err))
+	}
+
+	d.SetId(updateFg.Uuid)
+	return setFabricNetworkMap(d, updateFg)
+}
+
+func waitForFabricNetworkUpdateCompletion(uuid string, meta interface{}, ctx context.Context) (v4.Network, error) {
+	log.Printf("Waiting for Network update to complete, uuid %s", uuid)
+	stateConf := &resource.StateChangeConf{
+		Target: []string{string(v4.PROVISIONED_NetworkEquinixStatus)},
+		Refresh: func() (interface{}, string, error) {
+			client := meta.(*config.Config).FabricClient
+			dbConn, _, err := client.NetworksApi.GetNetworkByUuid(ctx, uuid)
+			if err != nil {
+				return "", "", equinix_errors.FormatFabricError(err)
+			}
+			return dbConn, string(*dbConn.Operation.EquinixStatus), nil
+		},
+		Timeout:    2 * time.Minute,
+		Delay:      30 * time.Second,
+		MinTimeout: 30 * time.Second,
+	}
+
+	inter, err := stateConf.WaitForStateContext(ctx)
+	dbConn := v4.Network{}
+
+	if err == nil {
+		dbConn = inter.(v4.Network)
+	}
+	return dbConn, equinix_errors.FormatFabricError(err)
```

Comment:
Double formatting here. We need to remove this. The Refresh part is the only part that needs to do the formatting for these errors.